### PR TITLE
Allow to provide the version to antsibull_core.args.get_toplevel_parser()

### PR DESCRIPTION
**Pull request by felixfontein**
Ref: https://github.com/ansible-community/antsibull-docs/pull/207

### CODE REVIEW

File: src/antsibull_core/args.py [hunk 1]
```diff
@@ -16,20 +16,27 @@ class InvalidArgumentError(Exception):
...
-        # If there's no metadata foun, assume we're running from source
-        version = "source"
+    if package_version is None:
+        try:
+            version = metadata.version(package)
+        except metadata.PackageNotFoundError:
+            # If there's no metadata foun, assume we're running from source
+            version = "source"
+    else:
+        version = package_version
```
Comment (gotmax23) [inline src/antsibull_core/args.py:39]: Is using an intermediate variable (both `version` and `package_version`) necessary?

Comment (felixfontein) [reply src/antsibull_core/args.py]: Nope. Fixed. I also noticed that it always says "Print the antsibull version", which is wrong for half of the users of antsibull-core. I've made that configurable as well.

File: src/antsibull_core/args.py [hunk 2]
```diff
@@ -16,27 +16,42 @@ class InvalidArgumentError(Exception):
...
     toplevel_parser = argparse.ArgumentParser(**kwargs)
     toplevel_parser.add_argument(
         "--version",
         action="version",
-        version=version,
-        help="Print the antsibull version",
+        version=package_version,
+        help=f"Print the {program_name} version"
+        if program_name
+        else "Print the program's version",
```
Comment (gotmax23) [inline src/antsibull_core/args.py:54]: Hmm, this `else` will only be reached if program_name is an empty string.

Comment (felixfontein) [reply src/antsibull_core/args.py:54]: It will also be reached if it is `None`, which is it's default value.

Comment (gotmax23) [reply src/antsibull_core/args.py:54]: You have

``` python
    if program_name is None:
        program_name = package
```

a couple lines above.

Comment (felixfontein) [reply src/antsibull_core/args.py:51]: Oops, I had that in for an earlier version and wanted to remove it. It's gone now.

File: src/antsibull_core/args.py [hunk 3]
```diff
@@ -16,27 +16,39 @@ class InvalidArgumentError(Exception):
...
-    try:
-        version = metadata.version(package)
-    except metadata.PackageNotFoundError:
-        # If there's no metadata foun, assume we're running from source
-        version = "source"
+    if package_version is None:
+        try:
+            package_version = metadata.version(package)
+        except metadata.PackageNotFoundError:
+            # If there's no metadata foun, assume we're running from source
```
Comment (gotmax23) [inline src/antsibull_core/args.py:41]: This was here before, but I guess we can fix it as part of this PR.

```suggestion
            # If there's no metadata found, assume we're running from source
```

**Review by gotmax23 — APPROVED** [review]
Thanks!

**Comment by felixfontein** [comment]
@gotmax23 thanks for reviewing!

**Pull request merged**